aeron-archive/src/main/java/io/aeron/archive/Archive.java

```diff
@@ -1963,34 +1994,49 @@ UnsafeBuffer dataBuffer()
         {
             if (null == dataBuffer)
             {
-                dataBuffer = new UnsafeBuffer(allocateDirectAligned(MAX_BLOCK_LENGTH, CACHE_LINE_LENGTH));
+                dataBuffer = allocateBuffer();
             }
-
             return dataBuffer;
         }
 
         UnsafeBuffer replayBuffer()
         {
-            if (null == replayBuffer)
+            if (DEDICATED != threadingMode)
```

Comment:
Why the extra checks on every access. Existing code pays price once.

Reply:
I want to void aliasing the same buffer from different variables so that close/free will be done only once. Of course one could do it by checking the threading mode in the `close`.